src/DynamoAnalyzer/Helper/GregHelper.cs

```diff
@@ -0,0 +1,241 @@
+            };
+
+            try
+            {
+
+                LogHelper.Log(packageHeader.name, $"{nameof(Download)}: start");
+                file = await Download(packageHeader._id, version.version);
+                LogHelper.Log(packageHeader.name, $"{nameof(Download)}: end");
+                LogHelper.Log(packageHeader.name, $"{nameof(Extract)}: start");
+                package = Extract(file);
+                LogHelper.Log(packageHeader.name, $"{nameof(Extract)}: end");
+                result.DLLs = await HasBin(package);
+            }
+            catch (Exception ex)
+            {
+                result.HasAnalysisError = true;
+                result.Result = new string[] { (ex.InnerException ?? ex).Message };
+                CleanWorkspaceOnError(file, package);
+            }
+
+            return result;
+        }
+
+        private static void CleanWorkspaceOnError(FileInfo file,
+        DirectoryInfo package)
+        {
+            try
+            {
+                if (file != null && file.Exists)
+                {
+                    file.Delete();
+                }
+
+                if (package != null && package.Exists)
+                {
+                    package.Delete(true);
+                }
+            }
+            catch (Exception)
+            {
+
+            }
+        }
+
+        /// <summary>
+        /// Download a package from dynamopackages.com
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="version"></param>
+        /// <returns></returns>
+        public static async Task<FileInfo> Download(string id, string version)
+        {
+            AnalyzeEnvironment.GetWorkspace().Create();
+
+            Response response = null;
+            string defaultDownloadPath = null;
+
+            await Task.Run(() =>
+            {
+                response = GetClient().Execute(new PackageDownload(id, version));
+                defaultDownloadPath = PackageDownload.GetFileFromResponse(response);
+            });
+
+            FileInfo fileInfo = new FileInfo(defaultDownloadPath);
+
+            string newFileLocation = Path.Combine(AnalyzeEnvironment.GetWorkspace().FullName, fileInfo.Name);
+
+            fileInfo.MoveTo(newFileLocation, true);
+            fileInfo = new FileInfo(newFileLocation);
+
+            return fileInfo;
+        }
+
+        /// <summary>
+        /// Extract a package archive to the workspace
+        /// </summary>
+        /// <param name="file"></param>
+        /// <returns></returns>
+        public static DirectoryInfo Extract(FileInfo file)
+        {
+            DirectoryInfo outputPath = new DirectoryInfo(GetPackageWorkspace(file));
+            if (outputPath.Exists)
+            {
+                outputPath.Delete(true);
+                outputPath.Refresh();
+            }
+
+            DirectoryInfo unzipPath = new DirectoryInfo(Greg.Utility.FileUtilities.UnZip(file.FullName));
```

Comment:
I think you can just reference the same base class that Greg does if you want to get rid of this reference - though Greg is already .net6 so should be fine.